app/javascript/components/cloud-networks/create-nsxt-cloud-network-form.jsx

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import MiqFormRenderer from '@@ddf';
+import createSchema from './create-nsxt-cloud-network-form.schema';
+import { CloudNetworkApi } from '../../utils/cloud-network-api'
+import { ProvidersApi } from '../../utils/providers.api';
+import { handleApiError } from '../../utils/handle-api-error';
+
+class CreateNsxtCloudNetworkForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { loading: true };
+  }
+
+  componentDidMount() {
+    this.props.dispatch({
+      type: 'FormButtons.init',
+      payload: {
+        newRecord: true,
+        pristine: true
+      }
+    });
+    this.setInitialState();
+  }
+
+  setInitialState = async () => {
+    miqSparkleOn();
+    try {
+      const nsxt_provider = await ProvidersApi.find_nsxt_provider();
```

Comment:
Instead of preselecting the provider, you can create a dropdown that dynamically loads a list of providers. This would fix the issue of having more than one NSX-T provider in your inventory. Also it would allow to make this component stateless as the loading would be sorted out by DDF. 

The [`loadOptions`](https://github.com/ManageIQ/manageiq-providers-redfish/blob/master/app/javascript/components/firmware-update.schema.js#L15) property on a select component is responsible for loading data dynamically. Its value should be a function that returns a [promise](https://github.com/ManageIQ/manageiq-providers-redfish/blob/master/app/javascript/components/redfish-server-firmware-update-dialog.jsx#L16) that loads the data in the same format as the `options` property would expect it.